auth_backends/backends.py

```diff
@@ -0,0 +1,114 @@
+"""Django authentication backends.
+
+For more information visit https://docs.djangoproject.com/en/dev/topics/auth/customizing/.
+"""
+import json
+
+from django.conf import settings
+import django.dispatch
+from social.backends.open_id import OpenIdConnectAuth
+
+
+# pylint: disable=abstract-method
+class EdXOpenIdConnect(OpenIdConnectAuth):
+    name = 'edx-oidc'
+
+    ACCESS_TOKEN_METHOD = 'POST'
+    REDIRECT_STATE = False
+    ID_KEY = 'preferred_username'
+
+    DEFAULT_SCOPE = ['openid', 'profile', 'email', 'permissions'] + getattr(settings, 'COURSE_PERMISSIONS_SCOPE', ['course_staff'])
+    ID_TOKEN_ISSUER = getattr(settings, 'SOCIAL_AUTH_EDX_OIDC_URL_ROOT', None)
+    AUTHORIZATION_URL = '{0}/authorize/'.format(ID_TOKEN_ISSUER)
+    ACCESS_TOKEN_URL = '{0}/access_token/'.format(ID_TOKEN_ISSUER)
+    USER_INFO_URL = '{0}/user_info/'.format(ID_TOKEN_ISSUER)
+
+    PROFILE_TO_DETAILS_KEY_MAP = {
+        'preferred_username': u'username',
+        'email': u'email',
+        'name': u'full_name',
+        'given_name': u'first_name',
+        'family_name': u'last_name',
+        'locale': u'language',
+    }
+
+    auth_complete_signal = django.dispatch.Signal(providing_args=["user", "id_token"])
+
+    def user_data(self, _access_token, *_args, **_kwargs):
+        # Include decoded id_token fields in user data.
+        return self.id_token
+
+    def auth_complete_params(self, state=None):
+        params = super(EdXOpenIdConnect, self).auth_complete_params(state)
+
+        # TODO: Due a limitation in the OIDC provider in the LMS, the list of all course permissions
+        # is computed during the authentication process. As an optimization, we explicitly request
+        # the list here, avoiding further roundtrips. This is no longer necessary once the limitation
+        # is resolved and instead the course permissions can be requested on a need to have basis,
+        # reducing overhead significantly.
+        claim_names = getattr(settings, 'COURSE_PERMISSIONS_CLAIMS', ['staff_courses'])
```

Comment:
Looks like this will leave a "claims" param with an empty id_token if there are no course_permissions_claims configured, is that alright?